test_themes/tests/test_crawl.py

```diff
@@ -51,4 +51,4 @@ def test_02_homepage_tour_every_theme(self):
         Website = self.env['website']
         websites_themes = Website.get_test_themes_websites()
         for website in websites_themes:
-            self.start_tour(f"/?fw={website.id}", 'homepage', login='admin')
+            self.start_tour(f"/web?fw={website.id}", 'homepage', login='admin')
```

Comment:
@rdeodoo I always forget: did not we drop the fw param? :grimacing:

Reply:
We did, it got deprecated (not used anymore but kept for stable) with https://github.com/odoo/odoo/commit/2c8c079c059f38b808c73bf17450a2efac2d388b and removed somewhere between 15.0 and master (propably with httpocalypse)

Reply:
But the `design-themes` `test_themes` module is adding the support for `fw` so it is fine I guess https://github.com/odoo/design-themes/blob/master/test_themes/models/ir_http.py#L16-L17

Reply:
Although, strange it would work on `/web` with an override to `_frontend_pre_dispatch(cls)` no @younn-o ?

Reply:
I changed it to _pre_dispatch for it to work

Reply:
That's a bit concerning.. It means it actually didn't work as expected and didn't change website, but tested the tour on the same default website in each loop?

Generally for other tests of that kind in `test_themes`, I wrote some step/check to ensure we landed on the expected website, eg
```py
self.assertTrue("/%s/static/src" % website.theme_id.name in res.text, "Ensure correct website was loaded")
```

That seems kinda important to fix/ensure here too, if what I said is correct.

Maybe the tour `homepage` (from website module) could be started with a URL query parameter passing the website to do the same check that I mentioned in JS? A new check step in the `homepage` tour checking that the asset URL contain the requested website.

A bit like on the `snippets_all_drag_and_drop` which check something from the URL parameter sent from the python.

Or any other idea, while it ensures we landed on the correct website

Reply:
About my previous remark, you can ignore it.
If that's true, that's a concern about stable/master too, the test is not robust enough and won't be red if the `fw` doesn't work anymore (meaning we actually don't really test anything).
I will have a look where the test was introduced, if that's indeed the case.

Reply:
I did a manual check (log the logo[aria-label] during the tour) and the websites are correctly changed during `test_02_homepage_tour_every_theme`
We could also maybe check on the html[data-website-id] ?
And indeed we should use the get_client_action_url method and not /web when starting the tours from python... I checked and for now, our runbot builds are a bit slower than master: 44 => 46 minutes for community and 62 => 66 minutes for enterprise, maybe we can improve with that :crossed_fingers: